### PR TITLE
ci: add workflow to build mo-service for each PR

### PR DESCRIPTION
**Pull request by lni**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [x] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #

## What this PR does / why we need it:

Config the CI workflow to build mo-service for each PR.

### CODE REVIEW

**Review by fengttt — COMMENTED** [review]
I think I missed this one.   Why do we need a mo-service?    What happened to our "single binary" story?

**Comment by lni** [comment]
> I think I missed this one. Why do we need a mo-service? What happened to our "single binary" story?

mo-service is the single binary. it can be configured via its toml configuration to play the role of CN, DN or Log node. 

the plan is to keep the current mo-server, which largely inherit its v0.5 behaviors, for near term future so we can still have all bvt executed in CI. when everything is ready in a later phase during the v0.6 cycle, we will atomically switch using a single PR to let all bvt to be executed on the new mo-service based implementation. of course we can later rename it to be mo-server. 

before the above mentioned switch, we need this workflow to ensure mo-service can always be built and this is the purpose of this PR.

**Review by fengttt — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -388,14 +388,21 @@ jobs:
           path: |
             ${{ github.workspace }}
           retention-days: 7
-          
+  
+  build-services:
```
Comment (sukki37) [inline .github/workflows/ci.yml:392]: If you want to test `make service`. You need refactor this job, a simple example:
```
  build-services:
    name: Build mo-service
    runs-on: ubuntu-latest
    steps:
      - name: checkout
        uses: actions/checkout@v2
        with:
          fetch-depth: '3'

      - name: Set up Go
        uses: ./.github/actions/setup_env

      - name: Build mo-service
        run: |
          cd $GITHUB_WORKSPACE && make service 
```

**Review by sukki37 — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (lni) [reply .github/workflows/ci.yml:392]: Thanks. Done.

**Review by sukki37 — APPROVED** [review]
lgtm

**Pull request merged**